api/odilo.py

```diff
@@ -0,0 +1,532 @@
+# coding=utf-8
+import datetime
+import json
+from flask.ext.babel import lazy_gettext as _
+
+from circulation import (
+    LoanInfo,
+    HoldInfo,
+    FulfillmentInfo,
+    BaseCirculationAPI,
+)
+from core.odilo import (
+    OdiloAPI as BaseOdiloAPI,
+    OdiloRepresentationExtractor,
+    OdiloBibliographicCoverageProvider,
+    MockOdiloAPI as BaseMockOdiloAPI,
+)
+
+from core.model import (
+    Credential,
+    DataSource,
+    DeliveryMechanism,
+    ExternalIntegration,
+    Identifier,
+    Representation,
+)
+
+from core.monitor import (
+    CollectionMonitor,
+)
+from core.util.http import HTTP
+
+from circulation_exceptions import *
+
+
+class OdiloAPI(BaseOdiloAPI, BaseCirculationAPI):
+    NAME = ExternalIntegration.ODILO
+    DESCRIPTION = _("Integrate an Odilo library collection.")
+    SETTINGS = [
+                   {"key": BaseOdiloAPI.LIBRARY_API_BASE_URL, "label": _("Library API base URL")},
+                   {"key": ExternalIntegration.USERNAME, "label": _("Client Key")},
+                   {"key": ExternalIntegration.PASSWORD, "label": _("Client Secret")},
+               ] + BaseCirculationAPI.SETTINGS
+
+    SET_DELIVERY_MECHANISM_AT = BaseCirculationAPI.FULFILL_STEP
+
+    # Create a lookup table between common DeliveryMechanism identifiers
+    # and Odilo format types.
+    epub = Representation.EPUB_MEDIA_TYPE
+    pdf = Representation.PDF_MEDIA_TYPE
+    mp3 = Representation.MP3_MEDIA_TYPE
+    mp4 = Representation.MP4_MEDIA_TYPE
+    wmv = Representation.WMV_MEDIA_TYPE
+    jpg = Representation.JPEG_MEDIA_TYPE
+    scorm = Representation.ZIP_MEDIA_TYPE
+
+    adobe_drm = DeliveryMechanism.ADOBE_DRM
+    no_drm = DeliveryMechanism.NO_DRM
+    streaming_text = DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+
+    # maps a 2-tuple (media_type, drm_mechanism) to the internal string used in Odilo API to describe that setup.
+    delivery_mechanism_to_internal_format = {
+        (pdf, adobe_drm): 'ACSM',
+        (pdf, streaming_text): 'PDF',
+        (epub, streaming_text): 'EPUB',
+        (mp3, no_drm): 'MP3',
+        (mp4, streaming_text): 'MP4',
+        (wmv, streaming_text): 'WMV',
+        (jpg, no_drm): 'JPG',
+        (scorm, no_drm): 'SCORM',
+    }
```

Comment:
You can get this by inverting `OdiloRepresentationExtractor.format_data_for_odilo_format`, and I recommend doing that rather than keep two copies of this information.

Reply:
Now that I know that a format like "EPUB" has two different delivery mechanisms, it might not work to simply invert the dictionary, but something similar should still work.

Reply:
I have also joined together both 'EPUB' and 'PDF' format constants just in one 'EBOOK_STREAMING' as they behave the same, and are basically the same thing for the final users.